### PR TITLE
Award Summary v2 Federal Accounts

### PR DESCRIPTION
**Pull request by ebdabbs**
**High level description:**
Implements the Federal Accounts section on award profile pages for non-IDVs.

**Technical details:**
- Moves federal account containers and components into the `shared` directories
- Also moves the `funding_rollup` table into its own container, since it makes a separate API call
- Removes accounts with negative obligated amounts while building the treemap, instead of making 2 API calls and passing down two sets of results that were parsed differently
- Updates the `ViewTypeButton` component to utilize FontAwesome icons
- Adds the helper file `awardSummaryHelper.js` for non-IDV API calls

**JIRA Ticket:**
[DEV-1662](https://federal-spending-transparency.atlassian.net/browse/DEV-1662)

**Mockup:**
https://bahdigital.invisionapp.com/share/U9IAAWH7MA6#/screens/296011639_Award_Summary_2-0_-_Contract

The following are ALL required for the PR to be merged:
- [x] Code review
- [x] Design review @AGuyNamedMarco 
- [x] [API contracts](https://github.com/fedspendingtransparency/usaspending-api/pull/1987) updated
- [x] [New API endpoints](https://github.com/fedspendingtransparency/usaspending-api/pull/2008) merged
- [x] [IDV endpoint updates](https://github.com/fedspendingtransparency/usaspending-api/pull/2011) merged 
- [x] Verified cross-browser compatibility

**Urls for testing**
Contract: https://sandbox.usaspending.gov/#/award_v2/12178133
Financial Assistance: https://sandbox.usaspending.gov/#/award_v2/82208056
IDV (no impact to end user): https://sandbox.usaspending.gov/#/award/68942216

### CODE REVIEW

File: src/js/components/awardv2/contract/ContractContent.jsx [hunk 1]
```diff
@@ -56,12 +56,16 @@ export default class ContractContent extends React.Component {
                     </AwardSection>
                 </AwardSection>
                 <AwardSection type="row">
-                    <AwardAmountsSection
-                        awardType={this.props.overview.category}
-                        jumpToSection={this.props.jumpToSection}
-                        awardOverview={this.props.overview}
-                        tooltipProps={defaultTooltipProps} />
-                    <ComingSoonSection title="Description" includeHeader />
```
Comment (MaxwellKendall) [inline src/js/components/awardv2/contract/ContractContent.jsx]: Lets keep the `<ComingSoonSection />` on line 64.

File: src/js/components/awardv2/financialAssistance/FinancialAssistanceContent.jsx [hunk 2]
```diff
@@ -43,8 +44,19 @@ export default class FinancialAssistanceContent extends React.Component {
             );
         }
 
-        const awardAmountData = Object.create(BaseAwardAmounts);
-        awardAmountData.populate(this.props.overview, this.props.overview.category);
+        let amountsSection = (<ComingSoonSection />);
+        if (this.props.overview.category === 'grant') {
```
Comment (MaxwellKendall) [inline src/js/components/awardv2/financialAssistance/FinancialAssistanceContent.jsx]: Thank you!! 💪

File: src/js/components/awardv2/financialAssistance/FinancialAssistanceContent.jsx [hunk 3]
```diff
@@ -73,13 +85,11 @@ export default class FinancialAssistanceContent extends React.Component {
...
-                            awardType={this.props.overview.category}
-                            awardOverview={awardAmountData}
-                            tooltipProps={defaultTooltipProps}
-                            jumptoSection={this.props.jumpToSection} />
+                        {amountsSection}
+                    </AwardSection>
+                    <AwardSection type="column">
+                        <FederalAccountsSection />
                     </AwardSection>
-                    <ComingSoonSection title="Description" includeHeader />
```
Comment (MaxwellKendall) [inline src/js/components/awardv2/financialAssistance/FinancialAssistanceContent.jsx]: Not sure why we are removing this?

File: src/js/containers/awardV2/shared/FederalAccountsSummaryContainer.jsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        if (prevProps.awardId !== this.props.awardId && (this.props.awardId && this.props.category)) {
+            await this.getAwardMetaData();
+        }
+    }
+
+    async getAwardMetaData() {
+        if (this.request) {
+            this.request.cancel();
+        }
+        if (this.props.category === 'idv') {
```
Comment (MaxwellKendall) [inline src/js/containers/awardV2/shared/FederalAccountsSummaryContainer.jsx]: FYSA I have a PR for the download button where I'm introducing a pattern where we abstract the condition into a new helper that takes the awardid and category as params and returns either the correct endpoint:

- [here's the helper that abstracts the conditional logic](https://github.com/fedspendingtransparency/usaspending-website/pull/1238/files?file-filters%5B%5D=.js#diff-2253e63fec011dd0d2986a9aa60409fbR113)
- [here's where its implemented](https://github.com/fedspendingtransparency/usaspending-website/pull/1238/files#diff-5d272f09dbb41cf6b50684bce394a07aR149)

 Bringing this to your attention but I think if we don't use this pattern here I should probably not use it in my pr for consistency sake?

File: src/js/components/awardv2/financialAssistance/FinancialAssistanceContent.jsx (hunk 3 above)
Comment (ebdabbs) [reply src/js/components/awardv2/financialAssistance/FinancialAssistanceContent.jsx]: I think this was in place of the column for the Federal Accounts section

File: src/js/containers/awardV2/shared/FederalAccountsSummaryContainer.jsx (hunk 4 above)
Comment (ebdabbs) [reply src/js/containers/awardV2/shared/FederalAccountsSummaryContainer.jsx]: Nice! I think that makes sense there, but here the API calls are in separate files (`idvHelper` and `awardSummaryHelper`) so I think it's ok if they're inconsistent.

File: src/js/components/awardv2/financialAssistance/FinancialAssistanceContent.jsx (hunk 3 above)
Comment (ebdabbs) [reply src/js/components/awardv2/financialAssistance/FinancialAssistanceContent.jsx]: Jk just checked the mockup and this was a placeholder for **Description**. I can add it back and make a new row to hold **Federal Accounts**

File: src/js/components/awardv2/contract/ContractContent.jsx [hunk 5]
```diff
@@ -56,12 +57,24 @@ export default class ContractContent extends React.Component {
...
+                    <AwardSection type="column">
+                        <AwardAmountsSection
+                            awardType={this.props.overview.category}
+                            jumpToSection={this.props.jumpToSection}
+                            awardOverview={this.props.overview}
+                            tooltipProps={defaultTooltipProps} />
+                    </AwardSection>
+                    <AwardSection type="column">
+                        <ComingSoonSection title="Description" includeHeader />
```
Comment (MaxwellKendall) [inline src/js/components/awardv2/contract/ContractContent.jsx]: The coming soon section already has the `<AwardSection type='column'>` in it so we should just do [this](https://github.com/fedspendingtransparency/usaspending-website/pull/1216/files?file-filters%5B%5D=.jsx&hide-deleted-files=true#diff-0563f5bfe1b06c58c0a566c0bbd040baL64)

File: src/js/components/awardv2/shared/federalAccounts/FederalAccountsTree.jsx [hunk 6]
```diff
@@ -47,12 +47,12 @@ export default class FederalAccountsTree extends React.Component {
         this.buildVirtualChart(this.props);
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (nextProps.data !== this.props.data) {
-            this.buildVirtualChart(nextProps);
+    componentDidUpdate(prevProps) {
```
Comment (MaxwellKendall) [inline src/js/components/awardv2/shared/federalAccounts/FederalAccountsTree.jsx]: nice

File: src/js/components/awardv2/financialAssistance/FinancialAssistanceContent.jsx (hunk 3 above)
Comment (MaxwellKendall) [reply src/js/components/awardv2/financialAssistance/FinancialAssistanceContent.jsx]: See comment [here](https://github.com/fedspendingtransparency/usaspending-website/pull/1216#discussion_r321779092) we shouldn't be nesting the `<ComingSoonSection />` in an `<AwardSection type='col' >`

**Comment by MaxwellKendall** [comment]
Two things:

1. Removing `AwardSection` parent to ComingSoon
2. The styling around the error chart still seems off for some reason. The font is too big?
image

**Comment by AGuyNamedMarco** [comment]
@ebdabbs @MaxwellKendall The treemap view in this branch shows only the current page's items. 
 The treemap view should show all items across all pages (as it is in production).

Screen Shot 2019-09-09 at 11 09 30 AM

**Comment by AGuyNamedMarco** [comment]
Thanks @ebdabbs.  Another thing I noticed is that when toggling between the tree and table views, the table doesn't toggled back to the page you were on (it's always brings you back to page 1).  

Can we retain the page when toggling back?

**Comment by millerjoey975** [comment]
> **High level description:**
> Implements the Federal Accounts section on award profile pages for non-IDVs.
> 
> **Technical details:**
> 
> * Moves federal account containers and components into the `shared` directories
> * Also moves the `funding_rollup` table into its own container, since it makes a separate API call
> * Removes accounts with negative obligated amounts while building the treemap, instead of making 2 API calls and passing down two sets of results that were parsed differently
> * Updates the `ViewTypeButton` component to utilize FontAwesome icons
> * Adds the helper file `awardSummaryHelper.js` for non-IDV API calls
> 
> **JIRA Ticket:**
> [DEV-1662](https://federal-spending-transparency.atlassian.net/browse/DEV-1662)
> 
> **Mockup:**
> https://bahdigital.invisionapp.com/share/U9IAAWH7MA6#/screens/296011639_Award_Summary_2-0_-_Contract
> 
> The following are ALL required for the PR to be merged:
> 
> * [x]  Code review
> * [x]  Design review @AGuyNamedMarco
> * [x]  [API contracts](https://github.com/fedspendingtransparency/usaspending-api/pull/1987) updated
> * [x]  [New API endpoints](https://github.com/fedspendingtransparency/usaspending-api/pull/2008) merged
> * [x]  [IDV endpoint updates](https://github.com/fedspendingtransparency/usaspending-api/pull/2011) merged
> * [x]  Verified cross-browser compatibility
> 
> **Urls for testing**
> Contract: https://sandbox.usaspending.gov/#/award_v2/12178133
> Financial Assistance: https://sandbox.usaspending.gov/#/award_v2/82208056
> IDV (no impact to end user): https://sandbox.usaspending.gov/#/award/68942216

**Pull request merged**